mmv1/products/identityplatform/Config.yaml

```diff
@@ -254,3 +254,116 @@ properties:
             description: |
               Two letter unicode region codes to allow as defined by https://cldr.unicode.org/ The full list of these region codes is here: https://github.com/unicode-cldr/cldr-localenames-full/blob/master/main/en/territories.json
             item_type: Api::Type::String
+  - !ruby/object:Api::Type::NestedObject
+    name: 'client'
+    description: |
+      Options related to how clients making requests on behalf of a project should be configured.
+    ignore_read: true
```

Comment:
I wonder what is the reason to add `ignore_read` ?

Reply:
Ignore read is added here because the API always returns the default values for multiple fields. 

```hcl
resource "google_identity_platform_config" "default" {
  project                    = "MYPROJECTID"
}
```

produces the following TF debug snippet:

```
2023-12-15T14:09:53.082Z [DEBUG] provider.terraform-provider-google: 2023/12/15 14:09:53 [DEBUG] Google API Request Details:
2023-12-15T14:09:53.082Z [DEBUG] provider.terraform-provider-google: ---[ REQUEST ]---------------------------------------
2023-12-15T14:09:53.082Z [DEBUG] provider.terraform-provider-google: GET /v2/projects/MYPROJECTID/config?alt=json HTTP/1.1
2023-12-15T14:09:53.082Z [DEBUG] provider.terraform-provider-google: Host: identitytoolkit.googleapis.com
2023-12-15T14:09:53.082Z [DEBUG] provider.terraform-provider-google: User-Agent: Terraform/1.6.1 (+https://www.terraform.io) Terraform-Plugin-SDK/2.10.1 terraform-provider-google/dev
2023-12-15T14:09:53.082Z [DEBUG] provider.terraform-provider-google: Content-Type: application/json
2023-12-15T14:09:53.082Z [DEBUG] provider.terraform-provider-google: X-Goog-User-Project: gleichda-xyz-demo
2023-12-15T14:09:53.082Z [DEBUG] provider.terraform-provider-google: Accept-Encoding: gzip
2023-12-15T14:09:53.082Z [DEBUG] provider.terraform-provider-google
2023-12-15T14:09:53.082Z [DEBUG] provider.terraform-provider-google
2023-12-15T14:09:53.082Z [DEBUG] provider.terraform-provider-google: -----------------------------------------------------
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google: 2023/12/15 14:09:53 [DEBUG] Google API Response Details:
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google: ---[ RESPONSE ]--------------------------------------
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google: HTTP/2.0 200 OK
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google: Cache-Control: no-cache, no-store, max-age=0, must-revalidate
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google: Content-Type: application/json; charset=UTF-8
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google: Date: Fri, 15 Dec 2023 14:09:53 GMT
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google: Expires: Mon, 01 Jan 1990 00:00:00 GMT
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google: Pragma: no-cache
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google: Server: ESF
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google: Vary: Origin
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google: Vary: X-Origin
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google: Vary: Referer
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google: X-Content-Type-Options: nosniff
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google: X-Frame-Options: SAMEORIGIN
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google: X-Xss-Protection: 0
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google: {
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:   "name": "projects/141423373225/config",
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:   "signIn": {
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:     "email": {
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:       "enabled": true
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:     },
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:     "phoneNumber": {
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:       "enabled": true,
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:       "testPhoneNumbers": {
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:         "+11231231234": "000000"
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:       }
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:     },
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:     "anonymous": {
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:       "enabled": true
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:     },
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:     "allowDuplicateEmails": true,
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:     "hashConfig": {
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:       "algorithm": "SCRYPT",
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:       "signerKey": "ocDgh1Kx2zrbQrzKZjKeQ5J0Cl4RALqSBBPi2+ZheygcJ75wsWc9uBWvokCWz8GrSLnhtEXfN+4s4S1Zzgz09A==",
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:       "saltSeparator": "Bw==",
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:       "rounds": 8,
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:       "memoryCost": 14
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:     }
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:   },
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:   "notification": {
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:     "sendEmail": {
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:       "method": "DEFAULT",
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:       "resetPasswordTemplate": {
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:         "senderLocalPart": "noreply",
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:         "subject": "Reset your password for %APP_NAME%",
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:         "body": "\u003cp\u003eHello,\u003c/p\u003e\n\u003cp\u003eFollow this link to reset your %APP_NAME% password for your %EMAIL% account.\u003c/p\u003e\n\u003cp\u003e\u003ca href='%LINK%'\u003e%LINK%\u003c/a\u003e\u003c/p\u003e\n\u003cp\u003eIf you didn’t ask to reset your password, you can ignore this email.\u003c/p\u003e\n\u003cp\u003eThanks,\u003c/p\u003e\n\u003cp\u003eYour %APP_NAME% team\u003c/p\u003e",
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:         "bodyFormat": "HTML",
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:         "replyTo": "noreply"
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:       },
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:       "verifyEmailTemplate": {
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:         "senderLocalPart": "noreply",
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:         "subject": "Verify your email for %APP_NAME%",
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:         "body": "\u003cp\u003eHello %DISPLAY_NAME%,\u003c/p\u003e\n\u003cp\u003eFollow this link to verify your email address.\u003c/p\u003e\n\u003cp\u003e\u003ca href='%LINK%'\u003e%LINK%\u003c/a\u003e\u003c/p\u003e\n\u003cp\u003eIf you didn’t ask to verify this address, you can ignore this email.\u003c/p\u003e\n\u003cp\u003eThanks,\u003c/p\u003e\n\u003cp\u003eYour %APP_NAME% team\u003c/p\u003e",
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:         "bodyFormat": "HTML",
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:         "replyTo": "noreply"
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:       },
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:       "changeEmailTemplate": {
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:         "senderLocalPart": "noreply",
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:         "subject": "Your sign-in email was changed for %APP_NAME%",
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:         "body": "\u003cp\u003eHello %DISPLAY_NAME%,\u003c/p\u003e\n\u003cp\u003eYour sign-in email for %APP_NAME% was changed to %NEW_EMAIL%.\u003c/p\u003e\n\u003cp\u003eIf you didn’t ask to change your email, follow this link to reset your sign-in email.\u003c/p\u003e\n\u003cp\u003e\u003ca href='%LINK%'\u003e%LINK%\u003c/a\u003e\u003c/p\u003e\n\u003cp\u003eThanks,\u003c/p\u003e\n\u003cp\u003eYour %APP_NAME% team\u003c/p\u003e",
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:         "bodyFormat": "HTML",
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:         "replyTo": "noreply"
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:       },
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:       "callbackUri": "https://MYPROJECTID.firebaseapp.com/__/auth/action",
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:       "dnsInfo": {
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:         "customDomainState": "NOT_STARTED",
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:         "domainVerificationRequestTime": "1970-01-01T00:00:00Z"
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:       },
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:       "revertSecondFactorAdditionTemplate": {
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:         "senderLocalPart": "noreply",
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:         "subject": "You've added 2 step verification to your %APP_NAME% account.",
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:         "body": "\u003cp\u003eHello %DISPLAY_NAME%,\u003c/p\u003e\n\u003cp\u003eYour account in %APP_NAME% has been updated with %SECOND_FACTOR% for 2-step verification.\u003c/p\u003e\n\u003cp\u003eIf you didn't add this 2-step verification, click the link below to remove it.\u003c/p\u003e\n\u003cp\u003e\u003ca href='%LINK%'\u003e%LINK%\u003c/a\u003e\u003c/p\u003e\n\u003cp\u003eThanks,\u003c/p\u003e\n\u003cp\u003eYour %APP_NAME% team\u003c/p\u003e",
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:         "bodyFormat": "HTML",
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:         "replyTo": "noreply"
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:       }
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:     },
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:     "sendSms": {
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:       "smsTemplate": {
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:         "content": "%LOGIN_CODE% is your verification code for %APP_NAME%."
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:       }
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:     },
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:     "defaultLocale": "en"
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:   },
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:   "quota": {},
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:   "monitoring": {
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:     "requestLogging": {}
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:   },
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:   "multiTenant": {},
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:   "authorizedDomains": [
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:     "localhost",
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:     "MYPROJECTID.firebaseapp.com",
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google: "MYPROJECTID.web.app"
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:   ],
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:   "subtype": "IDENTITY_PLATFORM",
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:   "client": {
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:     "apiKey": "AIzaSyDkF1gZrD1743YNYUMP8zwUWkEhS9z3yOY",
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:     "permissions": {},
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:     "firebaseSubdomain": "MYPROJECTID"
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:   },
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:   "mfa": {
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:     "state": "DISABLED"
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:   },
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:   "blockingFunctions": {},
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:   "smsRegionConfig": {},
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:   "autodeleteAnonymousUsers": true,
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:   "emailPrivacyConfig": {
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:     "enableImprovedEmailPrivacy": true
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google:   }
2023-12-15T14:09:53.312Z [DEBUG] provider.terraform-provider-google: }
2023-12-15T14:09:53.313Z [DEBUG] provider.terraform-provider-google
```

Reply:
Thanks for the information. Then `default_from_api` should be used here instead of `ignore_read`